graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -67,29 +83,36 @@
 
 
 def rename_schema(
-    schema_ast: DocumentNode, renamings: Mapping[str, str]
+    schema_ast: DocumentNode, renamings: Mapping[str, Optional[str]]
 ) -> RenamedSchemaDescriptor:
     """Create a RenamedSchemaDescriptor; types and query type fields are renamed using renamings.
 
     Any type, interface, enum, or fields of the root type/query type whose name
-    appears in renamings will be renamed to the corresponding value. Any such names that do not
-    appear in renamings will be unchanged. Scalars, directives, enum values, and fields not
-    belonging to the root/query type will never be renamed.
+    appears in renamings will be renamed to the corresponding value if the value is not None. If the
+    value is None, it will be suppressed in the renamed schema and queries will not be able to
+    access it.
+
+    Any such names that do not appear in renamings will be unchanged.
+
+    Scalars, directives, enum values, and fields not belonging to the root/query type will never be
+    renamed.
 
     Args:
         schema_ast: represents a valid schema that does not contain extensions, input object
                     definitions, mutations, or subscriptions, whose fields of the query type share
                     the same name as the types they query. Not modified by this function
-        renamings: maps original type/field names to renamed type/field names. Type or query type
-                   field names that do not appear in the dict will be unchanged. Any dict-like
-                   object that implements get(key, [default]) may also be used
+        renamings: maps original type name to renamed name or None (for type suppression). Any
+                   name not in the dict will be unchanged
 
     Returns:
-        RenamedSchemaDescriptor, a namedtuple that contains the AST of the renamed schema, and the
-        map of renamed type/field names to original names. Only renamed names will be included
-        in the map.
+        RenamedSchemaDescriptor containing the AST of the renamed schema, and the map of renamed
+        type/field names to original names. Only renamed names will be included in the map.
 
     Raises:
+        - CascadingSuppressionError if a type suppression would require further suppressions
```

Comment:
> Separately, it might be good to treat these "intermediate" error types as abstract, and always raise only errors whose types are not further refined.

Hmm, I don't quite understand this sentence-- what does it mean to "refine" a type here?

Reply:
All GraphQL compiler errors are subtypes of `GraphQLError`, but we never raise `GraphQLError` itself, instead opting for a subtype: any error the compiler raises passes `isinstance(e, GraphQLError)` but fails `type(e) == GraphQLError`.

I'm suggesting we could do the same for `SchemaTransformError`: have all exceptions raised from inside the `schema_transformation` module be derived from `SchemaTransformError` but always raise one of its subtypes instead of raising it directly. This is nice because then if you write `try ... except SchemaTransformError` you are explicitly stating that you intend to catch *all* such errors. Currently, since things can raise `SchemaTransformError` itself on its own, this is not true — there's no way to catch *just those errors* without also catching everything else, i.e. the intent behind that same `try ... except SchemaTransformError` would be unclear.

Reply:
Ah, got it. From our discussion a few weeks ago based on [this blog post](https://ericlippert.com/2008/09/10/vexing-exceptions/), at the time we decided to not make a new class since all the errors simply report an error message so a user wouldn't handle them differently. I do plan to update the errors so that the user can programmatically fix them, and at that point I think it makes sense to make a new class for this. For the time being, then, I'll just leave a note.